src/App.php

```diff
@@ -1623,7 +1617,409 @@ public function checkURL()
 		// We will only change the url to an ip address if there is no existing setting
 
 		if (empty($url) || (!link_compare($url, $this->getBaseURL())) && (!preg_match("/^(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})$/", $this->getHostName()))) {
-			Config::set('system', 'url', $this->getBaseURL());
+			Core\Config::set('system', 'url', $this->getBaseURL());
+		}
+	}
+
+	/**
+	 * Frontend App script
+	 *
+	 * The App object behaves like a container and a dispatcher at the same time, including a representation of the
+	 * request and a representation of the response.
+	 *
+	 * This probably should change to limit the size of this monster method.
+	 */
+	public function runFrontend()
+	{
+		// Missing DB connection: ERROR
+		if ($this->getMode()->has(App\Mode::LOCALCONFIGPRESENT) && !$this->getMode()->has(App\Mode::DBAVAILABLE)) {
+			Core\System::httpExit(500, ['title' => 'Error 500 - Internal Server Error', 'description' => 'Apologies but the website is unavailable at the moment.']);
+		}
+
+		// Max Load Average reached: ERROR
+		if ($this->isMaxProcessesReached() || $this->isMaxLoadReached()) {
+			header('Retry-After: 120');
+			header('Refresh: 120; url=' . $this->getBaseURL() . "/" . $this->query_string);
+
+			Core\System::httpExit(503, ['title' => 'Error 503 - Service Temporarily Unavailable', 'description' => 'Core\System is currently overloaded. Please try again later.']);
+		}
+
+		if (strstr($this->query_string, '.well-known/host-meta') && ($this->query_string != '.well-known/host-meta')) {
+			Core\System::httpExit(404);
+		}
+
+		if (!$this->getMode()->isInstall()) {
+			// Force SSL redirection
+			if (Core\Config::get('system', 'force_ssl') && ($this->getScheme() == "http")
+				&& intval(Core\Config::get('system', 'ssl_policy')) == SSL_POLICY_FULL
+				&& strpos($this->getBaseURL(), 'https://') === 0
+				&& $_SERVER['REQUEST_METHOD'] == 'GET') {
+				header('HTTP/1.1 302 Moved Temporarily');
```

Comment:
Don't we have some built mechanism for redirections?

Reply:
You must be talking about `System::httpExit()`.

Reply:
That isn't able to perform a redirect, isn't it?

Reply:
Not yet.